### PR TITLE
VC language normalization and presentation header support

### PR DESCRIPTION
**Pull request by quartzjer**
This updates the JPA draft to align with the updated language and terms in the main JWP draft as well as adds support for the presentation header to the SU and BBS algorithms.

It does NOT update the examples (yet), that is coming in a separate PR.

### CODE REVIEW

**Review by dwaite — APPROVED** [review]

File: draft-jmiller-json-proof-algorithms.md [hunk 1]
```diff
@@ -51,6 +51,8 @@ The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 document are to be interpreted as described in BCP 14 [@!RFC2119] [@RFC8174]
 when, and only when, they appear in all capitals, as shown here.
 
+The roles of "issuer", "holder", and "verifier", are used as defined by the [Verifiable Credentials Data Model v1.1](https://www.w3.org/TR/2021/REC-vc-data-model-20211109/).  The term "presentation" is also used as defined by this source, but the term "credential" is avoided in this specification in order to minimize confusion with other definitions.
```
Comment (OR13) [inline draft-jmiller-json-proof-algorithms.md:54]: +1 to this language.

File: draft-jmiller-json-proof-algorithms.md [hunk 2]
```diff
@@ -73,112 +79,127 @@ Due to the nature of ZKPs, this specification also documents the subtle but impo
...
 
 TODO:
 
-* MUST support the protected header as an octet string
+* MUST support the issuer protected header as an octet string
 * MUST support one or more payloads, each as an octet string
-* MAY support the output of the `request_signature` operation from the requesting party (for blinded payloads)
-* MUST include integrity protection for the header and all payloads
-* MUST specify all digest and hash2curve methods used
+* MAY support algorithm-specific options from the holder (for blinded payloads, replay prevention, etc)
```
Comment (OR13) [inline draft-jmiller-json-proof-algorithms.md:92]: +1 to this.

File: draft-jmiller-json-proof-algorithms.md [hunk 3]
```diff
@@ -73,112 +79,127 @@ Due to the nature of ZKPs, this specification also documents the subtle but impo
...
 
-Each individual JWS uses a fixed  protected header containing only the minimum required `alg` value.  Since this JWS protected header itself is the same for every JWS, it SHOULD be a static value in the form of `{"alg":"***"}` where `***` is the JWS asymmetric key algorithm being used.  This value is re-created by a Verifier using the correct algorithm value.
+If an implementation uses an alternative JWS protected header than this fixed value, a base64url encoded serialized form of the alternate fixed header MUST be included using the `jws_header` claim in the issuer protected header.
 
-If an implementation uses an alternative JWS protected header than the fixed value then a base64url encoded serialized form of that fixed header MUST be included as the `proof_header` value in the JWP protected header.
+### Issuer Protected Header
 
-### Protected Header
+The JWK of the issuer's Ephemeral Key MUST be included in the issuer protected header with the property name of `proof_jwk` and contain only the REQUIRED values to represent the public key.
```
Comment (OR13) [inline draft-jmiller-json-proof-algorithms.md:168]: hmm, i think `jwk` is already  a registered term for header use... I wonder if we need the proof prefix.

File: draft-jmiller-json-proof-algorithms.md [hunk 4]
```diff
@@ -73,112 +79,127 @@ Due to the nature of ZKPs, this specification also documents the subtle but impo
...
 
-The JWK of the ephemeral key MUST be included in the JWP protected header with the property name of `proof_jwk` and contain only the REQUIRED values to represent the public key.
+The holder's Replay Key JWK MUST be included in issuer protected header using the `presentation_jwk` claim.
```
Comment (OR13) [inline draft-jmiller-json-proof-algorithms.md:170]: ahh I see the reason to disambiguate here..

File: draft-jmiller-json-proof-algorithms.md [hunk 5]
```diff
@@ -202,39 +223,39 @@ In the implementation the method used to generate the key pairs is `generateBls1
...
 In the implementation, the method used to perform the signing is `blsSign({keyPair, [header, payload1, payload2, ...]})` and returns a binary signature value.
 
-### Proving
+### Presentation
 
-The prover must decode the JWP header and payload values in order to generate the identical message array that the signer used.
+The holder must decode the issuer protected header and payload values in order to generate the identical message array that the issuer used.
 
-To generate a JWP for a verifier, the prover must use a cryptographic nonce that is provided by that verifier as input.  This nonce MUST be a 32 byte octet string that the verifier generated by a secure RNG.
+To generate a presented JWP for a verifier, the holder must use a cryptographic nonce that is provided by that verifier as input.  This nonce MUST be a 32 byte octet string that the verifier generated by a secure RNG.  How this nonce value is communicated to the holder is out of scope of this presentation.  The `nonce` claim in the presentation protected header is used to store the verifier's given nonce value.
```
Comment (OR13) [inline draft-jmiller-json-proof-algorithms.md:246]: i like this language change.

**Review by selfissued — APPROVED** [review]
Looks like a good set of improvements to me.  Ship it!

**Pull request merged**